### DISCUSSION TITLE
Customizing django form widgets for a pretix payment plugin

### DISCUSSION THREAD

**Discussion by lassejaco** [pretix/pretix#2174] [Q&A] [answered]
Trying to customize the rendering of django RadioSelect widget; does anyone have any idea how I would go about this conceptually? seems easy enough in a standalone Django project, but I can't seem to "make the connection" between my widget templates in the pretix payment plugin and Django - I am a total django/python/pretix noob and I don't feel I have enough contextual awareness to get further easily on my own - hoping someone here has done something similar and can save me a headache or ten :-)

**Comment by raphaelm** [comment, marked as answer]
You can happily pass a `widget=` parameter to the form field and you can override the template on that widget, there shouldn't be anything pretix-specific about it. Can you maybe You can happily pass a `widget=` parameter to the form field and you can override the template on that widget, there shouldn't be anything pretix-specific about it. Can you maybe post snippets of what you tried already?

**Comment by lassejaco** [comment]
Hey, apologize for the radio silence - thanks for your answer!

Happy to hear that it's possible - conceptually I was on the right track it seems: I tried to override the template for the RadioSelect widget - I could not for the life of me get it to register. I don't have any code at hand - what would be immensely helpful would be an example of this working, but maybe I'm asking too much.

**Reply by raphaelm** [reply]
Yeah, you are asking too much, I guess ;) If you post your code that does not work and I spot an easy mistake I'm happy to point it out and provide any suggestions. But I'm not going to start coding an example, I'm not even more than 90 % sure what your goal is.

**Reply by lassejaco** [reply]
Totally fair, thanks for putting me on the right track regardless - we've compromised in the meantime, but perhaps I'll try again at a later time.

**Comment by rahul-kothari** [comment]
Hey! Found tis thread and was wondering if something like this is possible with pretix (and Django):

I have my own Pretix Payment provider class.

In `payment_form_fields(self)` method:
```
form_fields = OrderedDict(
            list(super().payment_form_fields.items())
            + [
                (
                    "currency_type",
                    forms.ChoiceField(
                        label=_("Payment currency"),
                        help_text=_("Select the currency you will use for payment."),
                        widget=forms.RadioSelect,
                        choices=currency_type_choices,
                        initial="ETH",
                    ),
                )
            ]
        )
```

`currency_type_choices` has several cryptocurrencies on different networks (e.g. Ethereum Mainnet, Ethereum Testnet or other blockchain networks.
Now based on the blockchain network it is in (can be determined by the name of the cryptocurrency), I wanted a separate widget.
        
I will admit I am fairly new to Django as well.
        
I really really appreciate any help you can provide here!

**Reply by raphaelm** [reply]
So you want a *dependency* between fields, i.e. a second widget changes based on the currency_type? That's not something Django can do, you will need to add all the fields and handle the display logic via JavaScript :\

**Reply by rahul-kothari** [reply]
Oof okay. Thanks for the quick reply!
